src/daft-core/src/utils/supertype.rs

```diff
@@ -195,6 +195,11 @@ pub fn get_supertype(l: &DataType, r: &DataType) -> Option<DataType> {
                 let inner_st = get_supertype(inner_left_dtype.as_ref(), inner_right_dtype.as_ref())?;
                 Some(DataType::List(Box::new(inner_st)))
             }
+            // TODO(Colin): Add support for getting supertype for two maps once StructArray supports such a cast.
```

Comment:
I think its fine to rm this todo. I can't imagine many binary operations that we'll require supertype logic for Map types